bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoBasicHandler.java

```diff
@@ -398,17 +415,24 @@ private boolean buildChannelStructure(String deviceName) {
             return null;
         }
         ChannelUID channelUID = new ChannelUID(getThing().getUID(), channel);
-        ChannelTypeUID channelTypeUID = new ChannelTypeUID(channelType);
 
         // TODO: Need to understand if this harms anything. If yes, channel only to be added when not there already.
         // current way allows to have no issues when channels are changing.
         if (getThing().getChannel(channel) != null) {
             logger.info("Channel '{}' for thing {} already exist... removing", channel, getThing().getUID());
             thingBuilder.withoutChannel(new ChannelUID(getThing().getUID(), channel));
         }
-        Channel newChannel = ChannelBuilder.create(channelUID, datatype).withType(channelTypeUID)
-                .withLabel(friendlyName).build();
-        thingBuilder.withChannel(newChannel);
+        ChannelBuilder newChannel = ChannelBuilder.create(channelUID, datatype).withLabel(friendlyName);
+        if (!channelType.isBlank()) {
+            ChannelTypeUID channelTypeUID = new ChannelTypeUID(channelType);
+            if (channelTypeRegistry.getChannelType(channelTypeUID) != null) {
```

Comment:
I'd imagine that channel types are contextual to the bindings that they are defined in, without the binding prefix I don't think that the registry would ever find the channel type you are looking for other than the system ones.

Reply:
True, but the binding prefix is added in the the getChannelType

https://github.com/openhab/openhab-addons/pull/8796/files#diff-1fbdf89cc35d86f5da2024be7a5588a90fca7e7314551a250755469e435ac900L138

Reply:
Ah, I saw the part you removed, but didn't notice the part you left in. You are correct.